### PR TITLE
Dark mode implemented for the search popup

### LINKED ISSUE #569: ✨ Enhancement: Dark mode for the search popup
### Is your feature request related to a problem? Please describe

No

### Describe the solution you'd like

Add dark mode for search popup 

output

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### Are you working on this?

No

### PR DESCRIPTION
**Pull request by crampussm**
**What kind of change does this PR introduce?**
Dark mode implemented on search popup. Previously the search modal (popup) was same in dark mode as it was in light mode. Now depending upon the mode it changes to light or dark.

**Issue Number:**

-  Closes #569  
-  Related to #___ 
-  Others? 

**Screenshots/videos:**

https://github.com/json-schema-org/website/assets/82692857/e7ce90a9-e40b-449e-beac-694bec4de1c6

**If relevant, did you update the documentation?**

**Summary**

**Does this PR introduce a breaking change?**
NO

### CODE REVIEW

**Comment by Sahil-Shadwal** [comment]
Nicely done @crampussm , I have a suggestion to make ,  look at the search of this https://react.dev/learn

Screenshot from 2024-03-22 12-12-37

It would look even better if the 
to select 
to navigate & esc button is white . Like in the SS.

**Review by Sahil-Shadwal — COMMENTED** [review]
Nicely done @crampussm , I have a suggestion to make , look at the search of react docs, here https://react.dev/learn

Screenshot from 2024-03-22 12-12-37


It would look even better if the
to select
to navigate & esc button is white . Like in the SS.

**Comment by crampussm** [comment]
> Nicely done @crampussm , I have a suggestion to make , look at the search of react docs, here https://react.dev/learn
> 
> Screenshot from 2024-03-22 12-12-37
> 
> It would look even better if the to select to navigate & esc button is white . Like in the SS.

Thank you for the suggestion. I would definitely do that.

**Comment by JeelRajodiya** [comment]
Hey! Do you have prettier extension installed in vscode? That might help with the failing workflow

**Comment by crampussm** [comment]
> Hey! Do you have prettier extension installed in vscode? That might help with the failing workflow

Thank you for helping me out

**Comment by crampussm** [comment]
I thing this is done. please review @benjagm

File: components/DarkModeToggle.tsx [hunk 1]
```diff
@@ -15,6 +15,20 @@ export default function DarkModeToggle() {
     const newTheme = isDarkMode ? 'light' : 'dark';
     setTheme(newTheme);
     setIsDarkMode(!isDarkMode);
+    const dataTheme = isDarkMode ? 'light' : 'dark';
+    const keyShadow = isDarkMode
```
Comment (benjagm) [inline components/DarkModeToggle.tsx:19]: Do we need to define this styles in both files: DarkModeToggle and Layout?

File: components/Layout.tsx [hunk 2]
```diff
@@ -152,10 +152,25 @@ const MainNavigation = () => {
   useEffect(() => {
     const icon = theme === 'dark' ? 'herobtn' : '';
     const menu = theme === 'dark' ? 'bg-white' : 'bg-black';
+    const dataTheme = theme === 'dark' ? 'dark' : 'light';
+    const keyShadow =
```
Comment (benjagm) [inline components/Layout.tsx:156]: How this logic differs from the one in DarkModeToggle.tsx?

**Review by benjagm — COMMENTED** [review]
Great work. It works but before merging I'd like to clarify what is the purpose of what seems to be some duplicated code.

File: components/DarkModeToggle.tsx (hunk 1 above)
Comment (crampussm) [reply components/DarkModeToggle.tsx]: Thank you for pointing out this. Actually it's only required in Layout file. We don't need it in DarkModeToggle file. I fixed it. Sorry for previous mistakes.

File: components/Layout.tsx (hunk 2 above)
Comment (crampussm) [reply components/Layout.tsx:156]: Actually we only need it here in Layout file not in DarkModeToggle.tsx, I have fixed it now.

**Comment by crampussm** [comment]
@benjagm please review. I made the fixes you told.

File: components/Layout.tsx [hunk 3]
```diff
@@ -152,10 +152,25 @@ const MainNavigation = () => {
...
+      theme === 'dark'
```
Comment (benjagm) [inline components/Layout.tsx:157]: Why not defining the styles as classes in the global css? That approach seems a cleaner way to manage styles.

**Review by benjagm — CHANGES_REQUESTED** [review]
It works thanks! I just left a comment to see if we clean the code moving the styles declaration to the css.

File: components/Layout.tsx (hunk 3 above)
Comment (crampussm) [reply components/Layout.tsx]: @benjagm did the changes you told, is it okay now? Please review.

**Comment by crampussm** [comment]
@benjagm did the changes you told, is it okay now? Please review.

**Review by benjagm — APPROVED** [review]
Looks great! Nice job!!

**Pull request merged**